File: Java/app/src/main/java/dev/ogabek/java/adapter/StoryAdapter.java
package dev.ogabek.java.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.imageview.ShapeableImageView;

import java.util.List;

import dev.ogabek.java.R;
import dev.ogabek.java.model.Story;

public class StoryAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    List<Story> stories;

    public StoryAdapter(List<Story> stories) {
        this.stories = stories;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_feed_story, parent, false);
        return new StoryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        Story story = stories.get(position);

        if (holder instanceof StoryViewHolder) {
            ((StoryViewHolder) holder).fullName.setText(story.getFullName());
            ((StoryViewHolder) holder).storyProfile.setImageResource(story.getProfile());
        }
    }

    @Override
    public int getItemCount() {
        return stories.size();
    }

    private static class StoryViewHolder extends RecyclerView.ViewHolder {

        ShapeableImageView storyProfile;
        TextView fullName;

        public StoryViewHolder(View view) {
            super(view);

            storyProfile = view.findViewById(R.id.iv_story_profile);
            fullName = view.findViewById(R.id.tv_story_full_name);

        }
    }
}

File: Kotlin/app/src/main/java/dev/ogabek/kotlin/adaptere/StoryAdapter.kt
package dev.ogabek.kotlin.adaptere

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.imageview.ShapeableImageView
import dev.ogabek.kotlin.R
import dev.ogabek.kotlin.model.Story

class StoryAdapter(var stories: List<Story>) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view: View =
            LayoutInflater.from(parent.context).inflate(R.layout.item_feed_story, parent, false)
        return StoryViewHolder(view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val story: Story = stories[position]
        if (holder is StoryViewHolder) {
            holder.fullName.text = story.fullName
            holder.storyProfile.setImageResource(story.profile)
        }
    }

    override fun getItemCount(): Int {
        return stories.size
    }

    private class StoryViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var storyProfile: ShapeableImageView = view.findViewById(R.id.iv_story_profile)
        var fullName: TextView = view.findViewById(R.id.tv_story_full_name)

    }

}